### ISSUE TITLE
Sik-Ho Tang | Review -- Stacked Denoising Autoencoders (Self-Supervised Learning).

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#118]
Sik-Ho Tang. [Review — Stacked Denoising Autoencoders (Self-Supervised Learning)](https://sh-tsang.medium.com/review-stacked-denoising-autoencoders-self-supervised-learning-c8ff81cef34c).

**Comment by NorbertZheng** [comment]
## Overview
One of the Earliest **Reconstruction-Based Self-Supervised Learning Approaches**, Using Denoising Autoencoders/Stacked Denoising Autoencoders.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/6d00b4ac-c5f8-4270-acc7-f7521e1d31e4)
Stacked Autoencoder (Figure from [Setting up stacked autoencoders](https://subscription.packtpub.com/book/big_data_and_business_intelligence/9781787121089/4/ch04lvl1sec51/setting-up-stacked-autoencoders)).

In this story, **Extracting and Composing Robust Features with Denoising Autoencoders**, (Denoising Autoencoders/Stacked Denoising Autoencoders), by Universite de Montreal, is briefly reviewed. This is a paper by Prof. Yoshua Bengio’s research group. In this paper:
 - Denoising Autoencoder is designed to reconstruct a denoised image from a noisy input image.
 - By training the denoising autoencoder, feature learning is achieved without using any labels, which is then used for **fine-tuning in image classification tasks**.
 - This paper should be one of the early papers for self-supervised learning.

This is a paper in 2008 ICML with over 5800 citations. And later published in 2010 JMLR with over 6200 citations.

**Comment by NorbertZheng** [comment]
## Denoising Autoencoder
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/533425c7-b349-4720-b19f-9f77aa803c21)
Denoising Autoencoder.

 - $x$: Original input image.
 - $\tilde{x}$: Corrupted image.
 - $y$: Hidden representation:

$$
y=f_{\theta}(\tilde{x})=s(W\tilde{x}+b),
$$

 - $z$: Reconstructed image.

$$
z=g_{\theta'}(y)=s(W'y+b'),
$$

Autoencoder consists of an encoder and a decoder.
 - Encoder: The corrupted input $\tilde{x}$ is first mapped to a hidden representation $y$.
 - Decoder: Then the cleaned input $z$ is reconstructed from $y$.

The above autoencoder only got one layer $f_{\theta}$ at encoder, and one $g_{\theta'}$ at decoder.

**Comment by NorbertZheng** [comment]
## Stacked Denoising Autoencoder
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/bd48a3f6-fa0e-4d8c-b72e-479f2f565ce4)
Stacking Denoising Autoencoders.

 - To train a deep autoencoder, at that time, it was difficult to train. The autoencoder is trained **layer-by-layer** at that moment.

Left: After training a first level denoising autoencoder (i.e. $f_{\theta}$ in the first figure), its learnt encoding function $f_{\theta}$ is used on clean input (left).

Middle: The resulting representation is used to train a second level denoising autoencoder to learn a second level encoding function $f_{\theta}^{(2)}$.

Right: From there, the procedure can be repeated to have deeper model.

**Comment by NorbertZheng** [comment]
## Fine-Tuning for Image Classification
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/2c55de34-c0bd-4779-a869-fd6f6da498d8)
Fine-tuning of a deep network for classification.

 - After training a stack of encoders as explained in the previous figure, an output layer is added on top of the stacked layers of the encoder part.

The parameters of the whole system are fine-tuned to minimize the error in predicting the supervised target (e.g., class), by performing gradient descent on a supervised cost.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/5d6c0aea-c5b7-4b20-bc96-cc1c015363e7)
Training and Fine-Tuning of an Autoencoder (Figure from [Setting up stacked autoencoders](https://subscription.packtpub.com/book/big_data_and_business_intelligence/9781787121089/4/ch04lvl1sec51/setting-up-stacked-autoencoders)).

The above figure shows the general steps for pre-training using autoencoder, and fine-tuning using encoder.

**Comment by NorbertZheng** [comment]
## Experimental Results
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/ae4c7f15-b663-47c4-8698-32aefa0dda0d)
Data sets (Characteristics of the 10 different problems considered).

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/312a2c90-73a3-4c09-9bae-5b509e942203)
Samples form the various image classification problems.

Different corruptions are added to the dataset for testing.
  - rot: Rotation.
  - bg-rand: Addition of a background composed of random pixels
  - bg-img: Addition of a background composed of patches extracted from a set of image, etc.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/76f710d7-8649-486e-8da4-4eed9df2cf7b)
Comparison of stacked denoising autoencoders (SDAE-3) with other models.

SDAE-3: Neural networks with 3 hidden layers initialized by stacking denoising autoencoders.

The encoder part is fine-tuned on the classification tasks.

SDAE-3 algorithm performs on par or better than the best other algorithms, including deep belief nets.

**Comment by NorbertZheng** [comment]
Unsupervised initialization of layers with an explicit denoising criterion helps to capture interesting structure in the input distribution.

This in turn leads to intermediate representations much better suited for subsequent learning tasks such as supervised classification.

**Comment by NorbertZheng** [comment]
## References
 - [2008 ICML] [Denoising Autoencoders] [Extracting and Composing Robust Features with Denoising Autoencoders](https://www.cs.toronto.edu/~larocheh/publications/icml-2008-denoising-autoencoders.pdf).
 - [2010 JMLR] [Stacked Denoising Autoencoders] [Stacked Denoising Autoencoders: Learning Useful Representations in a Deep Network with a Local Denoising Criterion](http://www.jmlr.org/papers/volume11/vincent10a/vincent10a.pdf).
